Dcr 11 auto assign1%rebate

amrarick26 opened the pull request.

amrarick26 committed:
DCR-11
Applies a 1% "rebate" to the order - promo found based on ID - will refactor to match rule expressions

amrarick26 committed:
DCR-11
Remove ability to remove promotion if promo is 1% rebate.

amrarick26 committed:
DCR-11
Find promo based on ValueExpression, not ID

amrarick26 committed:
Add ReadMe for addPromotion

amrarick26 committed:
DCR-11
Return toastr error if promo not found

amrarick26 committed:
Remove unused code

amrarick26 committed:
Remove unused code

crhistianramirez commented on src/app/addPromotion/addPromotion.js:
```diff
@@ -19,4 +20,27 @@ function AddPromotionComponentCtrl($exceptionHandler, $rootScope, OrderCloud, to
+                    return OrderCloud.Orders.AddPromotion(order.ID, promo.Code)
+                        .then(function() {
+                            $rootScope.$broadcast('OC:UpdateOrder', order.ID);
+                        })
```
I don't think this is the best solution here because we are limiting ourselves in terms of what promotion can be used in the future. 

We should pass in a specific ID and then have that ID live somewhere, possibly the catalog or if its a buyer specific promotion we can have it live on the buyer. 

By doing it this way, we can simply update the catalog with a new ID. Or if perhaps down the road when they decide they want to make it a 15% rebate we can just patch the promotion

amrarick26 committed:
DCR-11 work on refactoring auto applied rebate

amrarick26 committed:
DCR-11
Update AddReate service

amrarick26 committed:
DCR-11 add rootscope broadcast to promotion update

amrarick26 committed:
Merge branch 'development' of https://github.com/etundra-bsmith/angular-buyer into DCR-11_AutoAssign1%Rebate

amrarick26 committed:
DCR-11
Fix for promo not updating discount amount when LI qty changes

amrarick26 committed:
DCR-11
Clean up code

amrarick26 committed:
DCR-11
Set constant for promo (rebate) code
Fix for promo not showing up initially on cart

amrarick26 committed:
DCR-11
work on cart badge bug

amrarick26 committed:
DCR-11
Work on bugs for cart

amrarick26 committed:
DCR-11
Promo displayed correctly

amrarick26 committed:
DCR-11
Remove unused code off product-card

amrarick26 committed:
DCR-11
Remove defer promises in service
